src/pages/forecasts/ForecastsOverviewSection.tsx

```diff
@@ -0,0 +1,247 @@
+            <Button
+              startIcon="filter"
+              endIcon={!hasAccessToForecastsFeature ? 'sparkles' : undefined}
+              size="small"
+              variant="quaternary"
+              onClick={(e) => {
+                if (!hasAccessToForecastsFeature) {
+                  e.stopPropagation()
+                  premiumWarningDialogRef.current?.openDialog()
+                } else {
+                  onClick()
+                }
+              }}
+            >
+              {translate('text_66ab42d4ece7e6b7078993ad')}
+            </Button>
+          )}
+        >
+          <div className="flex flex-col gap-4">
+            <div className="flex items-center justify-between">
+              <div className="flex items-center gap-4">
+                <Typography variant="subhead1" color="grey700">
+                  {translate('text_1753014457040hhkl9fy58wy')}
+                </Typography>
+              </div>
+            </div>
+
+            <div className="flex w-full flex-col gap-3">
+              <Filters.Component />
+            </div>
+          </div>
+        </Filters.Provider>
+      </div>
+
+      {hasError && (
+        <GenericPlaceholder
+          title={translate('text_636d023ce11a9d038819b579')}
+          subtitle={translate('text_636d023ce11a9d038819b57b')}
+          image={<ErrorImage width="136" height="104" />}
+        />
+      )}
+
+      {!hasError && (
+        <AnalyticsStateProvider>
+          <MultipleLineChart
+            xAxisDataKey="startOfPeriodDt"
+            xAxisTickAttributes={['endOfPeriodDt', 'endOfPeriodDt']}
+            currency={selectedCurrency}
+            data={data}
+            loading={isLoading}
+            timeGranularity={timeGranularity}
+            lines={[
+              {
+                dataKey: 'amountCentsForecastOptimistic',
+                colorHex: FORECASTS_GRAPH_COLORS.amountCentsForecastOptimistic,
+                tooltipLabel: translate('text_17564701329935iz92g07zaj'),
+                strokeDasharray: '3 3',
+              },
+              {
+                dataKey: 'amountCentsForecastRealistic',
+                colorHex: FORECASTS_GRAPH_COLORS.amountCentsForecastRealistic,
+                tooltipLabel: translate('text_1756470132993ziruszj5lu1'),
+                strokeDasharray: '3 3',
+              },
+              {
+                dataKey: 'amountCentsForecastConservative',
+                colorHex: FORECASTS_GRAPH_COLORS.amountCentsForecastConservative,
+                tooltipLabel: translate('text_1756470132993wnnhcrw15q9'),
+                strokeDasharray: '3 3',
+              },
+            ]}
+          />
+
+          <HorizontalDataTable
+            leftColumnWidth={220}
+            columnWidth={timeGranularity === TimeGranularityEnum.Monthly ? 180 : 228}
+            data={data}
+            loading={isLoading}
+            initScrollTo={InitScrollTo.START}
+            rows={[
```

Comment:
Can you put it in a variable before the return? You may have to import the typing but I think we should start putting the long/complex configuration/logic outside of the templating for readability